src/Logger-Rohde&Schwarz_RTX/main.py

```diff
@@ -0,0 +1,252 @@
+# This Device Class is published under the terms of the MIT License.
+# Required Third Party Libraries, which are included in the Device Class
+# package for convenience purposes, may have a different license. You can
+# find those in the corresponding folders or contact the maintainer.
+#
+# MIT License
+#
+# Copyright (c) 2024 SweepMe! GmbH (sweep-me.net)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+# SweepMe! driver
+# * Module: Logger
+# * Instrument: Rohde&Schwarz RT-Series (RTA, RTB, ...)
+from __future__ import annotations
+
+from pysweepme.EmptyDeviceClass import EmptyDevice
+
+
+class Device(EmptyDevice):
+    """Logger driver to read out MEAS channels of Rohde&Schwarz Oscilloscope RT-Series (RTA, RTB, ...)."""
+    description = """
+                    <h3>Rohde & Schwarz Oscilloscope Meas Slots</h3>
+                    <p>This driver reads out the measurement 'places' of Rohde & Schwarz RT Oscilloscopes. Measurement
+                    places are measurements such as as Maximum peak height, periodicity, rise time, and much more. The
+                    measurement places can be definied on the device or via SweepMe! for given mode and source
+                    (e.g. Channel 1-4).
+                    <p>The driver has been tested with RTB2004, but should also work with other RT-Series devices.</p>
+                    </p>
+                    <p>Setup:</p>
+                    <ul>
+                    <li>Enable remote control at device.</li>
+                    <li>If use preset, the defined measurement places from the devices are used. Note that currently
+                    SweepMe! is unable to display the used measurement mode and units in preset mode.</li>
```

Comment:
Is that really a limitation of SweepMe! or rather this driver?

Reply:
The label and unit of the measurement, which correspond to self.variables and self.units, can only be requested after a connection to the device has been made. AFAIK SweepMe needs to know the variables and units before connect. I was unable to expand/change the variables afterwards